elections.js

```diff
@@ -6,41 +6,68 @@ const virtualChainId = 1100000;
+        const queryProcessingStarted = await client.createQuery(orbsVotingContractName, "hasProcessingStarted", []);
+        const responseProcessingStarted = await client.sendQuery(queryProcessingStarted);
 
-    const _isProcessingPeriod = await client.createQuery(orbsVotingContractName, "isProcessingPeriod", []);
-    const responseIsProcessingPeriod = await client.sendQuery(_isProcessingPeriod);
+        const _isProcessingPeriod = await client.createQuery(orbsVotingContractName, "isProcessingPeriod", []);
+        const responseIsProcessingPeriod = await client.sendQuery(_isProcessingPeriod);
 
-    const getCurrentEthereumBlockNumberQuery = await client.createQuery(orbsVotingContractName, "getCurrentEthereumBlockNumber", []);
-    const responsegetCurrentEthereumBlockNumber = await client.sendQuery(getCurrentEthereumBlockNumberQuery);
+        const getCurrentEthereumBlockNumberQuery = await client.createQuery(orbsVotingContractName, "getCurrentEthereumBlockNumber", []);
+        const responsegetCurrentEthereumBlockNumber = await client.sendQuery(getCurrentEthereumBlockNumberQuery);
 
-    const ethereumBlockNumberPerOrbs = Number(responsegetCurrentEthereumBlockNumber.outputArguments[0].value);
-    const isProcessingPeriod = Number(responseIsProcessingPeriod.outputArguments[0].value);
-    const inProgress = !(Number(responseProcessingStarted.outputArguments[0].value) === 0);
+        const ethereumBlockNumberPerOrbs = Number(responsegetCurrentEthereumBlockNumber.outputArguments[0].value);
+        const isProcessingPeriod = Number(responseIsProcessingPeriod.outputArguments[0].value);
+        const inProgress = !(Number(responseProcessingStarted.outputArguments[0].value) === 0);
 
-    const ethResp = await fetch('http://eth4.orbs.com:8080');
-    const eth = await ethResp.json();
+        const ethResp = await fetch('http://eth4.orbs.com:8080');
+        const eth = await ethResp.json();
 
-    return {
-        eth,
-        elections: {
-            ethereumBlockNumberPerOrbs,
-            isProcessingPeriod,
-            next: Number(responseElectionBlock.outputArguments[0].value),
-            inProgress,
-        }
-    };
+        return {
+            ok: true,
+            eth,
+            elections: {
+                ethereumBlockNumberPerOrbs,
+                isProcessingPeriod,
+                next: Number(responseElectionBlock.outputArguments[0].value),
+                inProgress,
+            }
+        };
+    } catch (err) {
+        console.error(`There has been a problem getting the latest elections status due to an error: ${JSON.stringify(err)}`);
+        return {
+            ok: false,
+            err,
+        };
+    }
 }
 
-(async () => {
-    console.log(await getElectionsStatus());
-})();
+const poller = async () => {
+    let result = await getServerElections();
+    if (result.ok) {
+        currentElectionsData = Object.assign({}, result);
```

Comment:
What do we want the UI to show when the results are stale?
Since the UI already demands progress of the timestamp we will eventually get a red indication - but the user cannot distinguish between a polling problem and actual lagging in the Ethereum node. 
I would add a timestamp to the polling data like `sampleTs`. You can add it here in the `assign` call. 
Later we can (in a seperate PR) add a special indication in the UI for when this timestamp is old.

Reply:
Agreed, I'll add it.